data_pull.py

```diff
@@ -53,33 +81,158 @@ def pull_data(self, search_criteria: str, delete_csv: bool = False):
         options = Options()
         options.add_argument("--headless")
 
-        # Define directory where data will be downloaded and file type
-        dirname = os.path.abspath(os.path.dirname(__file__))
-        download_dir = os.path.join(dirname, "data", "")
-        file_type = "*.csv"
-
         # Open Chrome web browser
         driver = webdriver.Chrome(service=service, options=options)
 
+        # Define directory where data will be downloaded and file type
+        dirname = os.path.abspath(os.path.dirname(__file__))
+        download_dir = os.path.join(dirname, 'data', '')
+
         # Enable Chromium driver to download files while in headless mode
-        params = {"behavior": "allow", "downloadPath": download_dir}
-        driver.execute_cdp_cmd("Page.setDownloadBehavior", params)
+        params = {'behavior': 'allow', 'downloadPath': download_dir}
+        driver.execute_cdp_cmd('Page.setDownloadBehavior', params)
 
         # Navigate to Redfin website
-        driver.get("http://www.redfin.com")
-        driver.implicitly_wait(1)
+        driver.get('http://www.redfin.com')
+        time.sleep(1.5)
```

Comment:
I had done a little reading on the different kinds of waits a while back but I'll have to read them again. I remember thinking the implicit wait was what made sense to me in general, so I had tried using that but it didn't seem to be doing anything within the script. I was running into an issue where it would try to download the CSV before the actual action from selecting a filter was done being applied, so I would get incorrect data. Switching to time.sleep() fixed that issue for me, though I knew it wasn't very elegant. Let me do some more reading on the Fluent and Explicit waits and see if I can make this better.